### PR TITLE
Create face-detection package from face-landmarks-detection

### PR DESCRIPTION
**Pull request by ahmedsabie**
This is the first stage of splitting the face-landmarks-detection package into a face-detector and a face-mesh package

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/965)

### CODE REVIEW

**Review by lina128 — APPROVED** [review]
Looks great, thank you Ahmed! I left some minor comments.

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/965#-:-MxahB9E2O0sytlDQ6IO:baqosrs)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ahmedsabie)

___
*[face-detection/src/types.ts, line 55 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/965#-MxafKry9lFZv-mEfaVk:-MxafKry9lFZv-mEfaVl:b239psk) ([raw file](https://github.com/tensorflow/tfjs-models/blob/3cadfeda51d52ce3a76354c14e38246d42d1dd10/face-detection/src/types.ts#L55)):*
> ```XML
> 
> export interface Face {
>   keypoints: Keypoint[];
> ```

Add an annotation for the keypoints.



_Code quote:_
```XML
keypoints: Keypoint[];
```
___
*[face-detection/src/tfjs/constants.ts, line 22 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/965#-MxagLifDBmVgMjJK7Xv:-MxagLifDBmVgMjJK7Xw:bg9lu8e) ([raw file](https://github.com/tensorflow/tfjs-models/blob/3cadfeda51d52ce3a76354c14e38246d42d1dd10/face-detection/src/tfjs/constants.ts#L22)):*
> ```XML
> import {MediaPipeFaceDetectorTfjsEstimationConfig, MediaPipeFaceDetectorTfjsModelConfig} from './types';
> 
> export const DEFAULT_DETECTOR_MODEL_URL_FULL =
> ```

Do we use full ranged non-sparsed model? So we provide three options? In the readme, it says two options.



_Code quote:_
```XML
DEFAULT_DETECTOR_MODEL_URL_FULL 
```

**Review by lina128 — APPROVED** [review]
:lgtm_strong:

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/965#-:-MxahM_I5sMOO3Op6P88:b-jbsf8z)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ahmedsabie, @bazarevsky, and @ivan-grishchenko)

**Review by ahmedsabie — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/965)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @bazarevsky, @ivan-grishchenko, and @lina128)

___
*[face-detection/src/types.ts, line 55 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/965#-MxafKry9lFZv-mEfaVk:-MxeyQsPENYjuB8ouMf4:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-models/blob/3cadfeda51d52ce3a76354c14e38246d42d1dd10/face-detection/src/types.ts#L55)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Add an annotation for the keypoints.
</blockquote></details>

Done.
___
*[face-detection/src/tfjs/constants.ts, line 22 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/965#-MxagLifDBmVgMjJK7Xv:-MxeyRNsEHHy3r169TXt:b-q9q8rz) ([raw file](https://github.com/tensorflow/tfjs-models/blob/3cadfeda51d52ce3a76354c14e38246d42d1dd10/face-detection/src/tfjs/constants.ts#L22)):*
<details><summary><i>Previously, lina128 (Na Li) wrote…</i></summary><blockquote>

Do we use full ranged non-sparsed model? So we provide three options? In the readme, it says two options.
</blockquote></details>

I commented it out if needed in the future, it exists in the MediaPipe code but is not used, only sparse is

**Review by lina128 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/965)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ahmedsabie, @bazarevsky, @ivan-grishchenko, and @lina128)

___
*[face-detection/src/types.ts, line 56 at r2](https://reviewable.io/reviews/tensorflow/tfjs-models/965#-Mxf1XHs-Ur7PILLy8-g:-Mxf1XHs-Ur7PILLy8-h:b-huipmf) ([raw file](https://github.com/tensorflow/tfjs-models/blob/4c814e578dc6155f61d9d1124815286f19ee2c0e/face-detection/src/types.ts#L56)):*
> ```XML
> export interface Face {
>   box: BoundingBox;       // A bounding box around the detected face.
>   keypoints: Keypoint[];  // Points of interest in the detected face, such as
> ```

How many keypoints?



_Code quote:_
```XML
  keypoints: Keypoint[];  // Points of interest in the detected face, such as
                          // nose, eyes etc.
```
___
*[face-detection/src/tfjs/constants.ts, line 22 at r1](https://reviewable.io/reviews/tensorflow/tfjs-models/965#-MxagLifDBmVgMjJK7Xv:-Mxf1__KDJfMdn-6xIzw:b-ucfkn4) ([raw file](https://github.com/tensorflow/tfjs-models/blob/3cadfeda51d52ce3a76354c14e38246d42d1dd10/face-detection/src/tfjs/constants.ts#L22)):*
<details><summary><i>Previously, ahmedsabie wrote…</i></summary><blockquote>

I commented it out if needed in the future, it exists in the MediaPipe code but is not used, only sparse is
</blockquote></details>

Thanks.

File: face-detection/src/mediapipe/detector.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          this.faces.push({
+            keypoints: this.translateOutput(detection.landmarks),
+            box: getBoundingBox(detection.boundingBox)
+          });
+        }
+      }
+    });
+  }
+
+  private translateOutput(landmarks: faceDetection.NormalizedLandmarkList) {
```
Comment (ivan-grishchenko) [inline face-detection/src/mediapipe/detector.ts:71]: It's not exactly "translate". You are converting normalized coordinates to absolute.

**Review by ivan-grishchenko — APPROVED** [review]

**Review by bazarevsky — APPROVED** [review]

**Pull request merged**